chain/chain/src/types.rs

```diff
@@ -348,8 +341,7 @@ pub trait RuntimeAdapter: Send + Sync {
     ) -> Result<(), Box<dyn std::error::Error>>;
 
     /// Should be executed after accepting all the parts.
-    /// Returns `true` if state is set successfully.
-    fn confirm_state(&self, state_root: &StateRoot) -> Result<bool, Error>;
+    fn confirm_state(&self, state_root: &StateRoot) -> Result<(), Box<dyn std::error::Error>>;
```

Comment:
Why we don't use typed errors here? @mikhailOK

Reply:
Fixed by @mikhailOK